pkg/controller/vitessshard/vitessshard_controller.go

```diff
@@ -50,6 +50,7 @@ const (
 var (
 	maxConcurrentReconciles = flag.Int("vitessshard_concurrent_reconciles", 10, "the maximum number of different vitessshards to reconcile concurrently")
 	resyncPeriod            = flag.Duration("vitessshard_resync_period", 30*time.Second, "reconcile vitessshards with this period even if no Kubernetes events occur")
+	onlineFileSystemExpansion = flag.Bool("enable_online_fs_expansion", true, "if true, pod referencing the resized volume do not need to be restarted, but provided that the volume plug-in supports, such as GCE-PD, AWS-EBS, Cinder, and Ceph RBD")
```

Comment:
IMO we should make this behavior opt-in because we have no way to know that this will work in a given environment, right? That would mean  making the default `false`. I would recommend:
```
onlineFileSystemExpansion = flag.Bool("enable_online_fs_expansion", false, "if true, pods referencing the resized volume do not need to be restarted; you must ensure that the CSI driver(s) in use support online resizing, e.g. GCE-PD, AWS-EBS, Cinder, or Ceph RBD")
```

Reply:
We can at least wait 2 releases before changing the default